rcl/test/rcl/test_lexer_lookahead.cpp

```diff
@@ -0,0 +1,317 @@
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+    EXPECT_EQ(RCL_LEXEME_EOF, lexeme1);
+    EXPECT_EQ(RCL_LEXEME_EOF, lexeme2);
+  }
+  // Accepting keeps the lexer at EOF
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_lexer_lookahead2_accept(&buffer, NULL, NULL));
+    rcl_lexeme_t lexeme = RCL_LEXEME_NONE;
+    ret = rcl_lexer_lookahead2_peek(&buffer, &lexeme);
+    EXPECT_EQ(RCL_RET_OK, ret);
+    EXPECT_EQ(RCL_LEXEME_EOF, lexeme);
+  }
+}
+
+
+TEST_F(CLASSNAME(TestLexerLookaheadFixture, RMW_IMPLEMENTATION), test_accept)
+{
+  rcl_ret_t ret;
+  rcl_lexer_lookahead2_t buffer;
+  SCOPE_LOOKAHEAD2(buffer, "foobar/");
+
+  rcl_lexeme_t lexeme = RCL_LEXEME_NONE;
+  const char * lexeme_text;
+  size_t lexeme_text_length;
+
+  // Peek token
+  ret = rcl_lexer_lookahead2_peek(&buffer, &lexeme);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_EQ(RCL_LEXEME_TOKEN, lexeme);
+
+  // accept token
+  ret = rcl_lexer_lookahead2_accept(&buffer, &lexeme_text, &lexeme_text_length);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_STREQ("foobar", std::string(lexeme_text, lexeme_text_length).c_str());
+
+  // peek forward slash
+  ret = rcl_lexer_lookahead2_peek(&buffer, &lexeme);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_EQ(RCL_LEXEME_FORWARD_SLASH, lexeme);
+
+  // accept forward slash
+  ret = rcl_lexer_lookahead2_accept(&buffer, &lexeme_text, &lexeme_text_length);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_STREQ("/", std::string(lexeme_text, lexeme_text_length).c_str());
+
+  // peek eof
+  ret = rcl_lexer_lookahead2_peek(&buffer, &lexeme);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_EQ(RCL_LEXEME_EOF, lexeme);
+
+  // accept eof
+  ret = rcl_lexer_lookahead2_accept(&buffer, &lexeme_text, &lexeme_text_length);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_STREQ("", std::string(lexeme_text, lexeme_text_length).c_str());
+
+  // peek eof again
+  ret = rcl_lexer_lookahead2_peek(&buffer, &lexeme);
+  EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_EQ(RCL_LEXEME_EOF, lexeme);
+}
+
+TEST_F(CLASSNAME(TestLexerLookaheadFixture, RMW_IMPLEMENTATION), test_expect)
+{
+  rcl_ret_t ret;
+  rcl_lexer_lookahead2_t buffer;
+  SCOPE_LOOKAHEAD2(buffer, "node_name:__node:=new_1");
+  const char * lexeme_text;
+  size_t lexeme_text_length;
+
+  ret = rcl_lexer_lookahead2_expect(&buffer, RCL_LEXEME_TOKEN, &lexeme_text, &lexeme_text_length);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
+  EXPECT_STREQ("node_name", std::string(lexeme_text, lexeme_text_length).c_str());
+
+  ret = rcl_lexer_lookahead2_expect(
+    &buffer, RCL_LEXEME_FORWARD_SLASH, &lexeme_text, &lexeme_text_length);
+  EXPECT_EQ(RCL_RET_WRONG_LEXEME, ret) << rcl_get_error_string_safe();
+}
+
+TEST_F(CLASSNAME(TestLexerLookaheadFixture, RMW_IMPLEMENTATION), test_lex_long_string)
```

Comment:
Should we add some additional test cases to cover things that are in the design doc, or does this one cover enough that it would be unnecessary?

Reply:
http://design.ros2.org/articles/static_remapping.html#applications-of-the-syntax

Reply:
Added tests using remapping rules from design doc in b408d2b4ff3a71bb3ac7d5c7e5fe9916791405a1